geometry/r3_element.hpp

```diff
@@ -23,41 +25,74 @@ using quantities::Quotient;
 template<typename Scalar>
 struct SphericalCoordinates;
 
-// An |R3Element<Scalar>| is an element of Scalar³. |Scalar| should be a vector
-// space over ℝ, represented by |double|. |R3Element| is the underlying data
-// type for more advanced strongly typed structures suchas |Multivector|.
-template<typename Scalar>
+template<typename ScalarX,
+         typename ScalarY = ScalarX,
+         typename ScalarZ = ScalarX>
+struct R3Element;
+
+template<typename T>
+struct enable_if_normed : not_constructible {};
+
+template<typename T>
+struct enable_if_normed<R3Element<T, T, T>> : not_constructible {
+  using Scalar = T;
+};
+
```

Comment:
Add a type alias (`norm_type<T>` or somesuch) for `typename enable_if_normed<T>::Scalar`, and use it below.